integrationtest/helper.go

```diff
@@ -0,0 +1,163 @@
+package integrationtest
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/taskcluster/slugid-go/slugid"
+	tcclient "github.com/taskcluster/taskcluster-client-go"
+	"github.com/taskcluster/taskcluster-client-go/queue"
+	"github.com/taskcluster/taskcluster-worker/commands"
+	_ "github.com/taskcluster/taskcluster-worker/commands/work"
+	_ "github.com/taskcluster/taskcluster-worker/config/env"
+	_ "github.com/taskcluster/taskcluster-worker/config/secrets"
+	_ "github.com/taskcluster/taskcluster-worker/engines/native"
+)
+
+var (
+	// all tests can share taskGroupId so we can view all test tasks in same
+	// graph later for troubleshooting
+	taskGroupID string = slugid.Nice()
+	testdata    string = testdataDir()
+)
+
+func testdataDir() string {
+	// some basic setup...
+	cwd, err := os.Getwd()
+	if err != nil {
+		panic(fmt.Errorf("Could not get current directory during test package initialisation: %v", err))
+	}
+	return filepath.Join(cwd, "testdata")
+}
+
+type TaskPayload struct {
+
+	// Artifacts to be published
+	Artifacts []struct {
+		Expires tcclient.Time `json:"expires,omitempty"`
+
+		// This will be the leading path to directories and the full name
+		// for files that are uploaded to s3. It must not begin or end
+		// with "/" and must only contain printable ascii characters
+		// otherwise.
+		//
+		// Syntax:     ^([\x20-\x2e\x30-\x7e][\x20-\x7e]*)[\x20-\x2e\x30-\x7e]$
+		Name string `json:"name"`
+
+		// File system path of artifact
+		//
+		// Syntax:     ^.*[^/]$
+		Path string `json:"path"`
+
+		// Artifacts can be either an individual `file` or a `directory` containing potentially multiple files with recursively included subdirectories
+		//
+		// Possible values:
+		//   * "file"
+		//   * "directory"
+		Type string `json:"type"`
+	} `json:"artifacts,omitempty"`
+
+	// Command to execute
+	Command []string `json:"command"`
+
+	// Optional URL for a gzipped tar-ball to downloaded and extracted in the HOME directory for running the command.
+	Context string `json:"context,omitempty"`
+
+	// Mapping from environment variables to values
+	Env json.RawMessage `json:"env,omitempty"`
+
+	// If true, reboot the machine after task is finished.
+	Reboot bool `json:"reboot,omitempty"`
+}
+
+func RunTestWorker(workerType string) {
+	os.Setenv("TASKCLUSTER_CAPACITY", "1")
+	os.Setenv("TASKCLUSTER_WORKER_TYPE", workerType)
+	os.Setenv("TASKCLUSTER_WORKER_ID", workerType)
+	os.Setenv("TASKCLUSTER_MAX_TASKS", "1")
```

Comment:
Obviously, we would then have to add them to the config, but then we wouldn't have to run against production queue.